### ISSUE TITLE
DataStreamer ContinuouslyAutoscale

### ISSUE THREAD

**Issue by rubenslkirchner** [ScottPlot/ScottPlot#3732]
**Issue:** If the ContinuouslyAutoscale is True some chart components get lost (Markers for example)

**ScottPlot Version:** 5.0.28

**Code Sample:** See http://scottplot.net/faq/repro/ for tips about creating reproducible code samples

```cs
using System;
using System.Collections.Generic;
using System.Threading;
using System.Threading.Tasks;
using ScottPlot;
using ScottPlot.Plottables;
using ScottPlot.WPF;

namespace trips.playback.plotters.scottplot;

public class RealTimeChart : ITestScenario
{
    private List<Task> _tasks;

    public RealTimeChart(WpfPlot wpfplot1, WpfPlot wpfplot2)
    {
        wpfplot1.Plot.Axes.AutoScale();
        wpfplot2.Plot.Axes.AutoScale();

        Task.Run(() => DoWork(wpfplot1));
    }

    private Task DoWork(WpfPlot plot)
    {
        const int points = 1000;

        var streamer = plot.Plot.Add.DataStreamer(points);
        //streamer.ContinuouslyAutoscale = true; // UNCOMMENT THIS LINE AND THE MARKER POSITION (X) GOES CRAZY. 
        streamer.ViewScrollLeft();

        int x = 0;
        Marker? m1 = null;

        while (true)
        {
            double y = Math.Sin(x);
            streamer.Add(y);

            if (m1 == null)
            {
                m1 = plot.Plot.Add.Marker(points, Math.Sin(x), MarkerShape.FilledCircle, 10, Colors.Black);
            }
            else
            {
                if (m1.Location.X > 0)
                {
                    m1.Location = new Coordinates(m1.Location.X - 1, m1.Location.Y);
                }
                else
                {
                    m1.Location = new Coordinates(points, y);
                }
            }

            Thread.Sleep(10);
            plot.Refresh();
            x++;
        }
    }
}
```

**Comment by rubenslkirchner** [comment]
Wrong behaviour

![trips playback plotters scottplot_YRISfYmTmr](https://github.com/ScottPlot/ScottPlot/assets/7251969/238af421-eb35-4ba3-8e95-0a6fb2c8c99c)

**Comment by swharden** [comment]
Hi @rubenslkirchner, I'm starting to work on this now. Can you clarify what behavior you expect? If the data streamer is only 1000 points and new data causes that marker to roll off the screen to the left, do you want the autoscaling to continue to widen the view to include it in the plot area even though it will just look like a marker floating in space with no data behind it?

Either way I'll refactor the "ContinuouslyAutoscale" feature a bit, moving it into the `Plot` itself, allowing it to be used for all data types.

**Comment by rubenslkirchner** [comment]
@swharden good news! The problem is that the marker is floating over the line series. Did you notice this?

**Comment by swharden** [comment]
I noticed it, but I'm not sure I understand the expected behavior. I'm about to reply with a screenshot to see if it's what you had in mind...

**Comment by rubenslkirchner** [comment]
Right at the beginning of the video, the space between the marker and the sine values ​​gets closer but should maintain the same distance.

**Comment by swharden** [comment]
As an alternative to allowing a single `DataStreamer` to manage the axes, I'm adding a flag to autoscale the plot (evaluating all plotted objects) at the start of each render.

```cs
WpfPlot1.Plot.Axes.ContinuouslyAutoscale = true;
```

The video shows what happens when you switch between the two control strategies. What do you think about the behavior displayed in this video?

![mark](https://github.com/ScottPlot/ScottPlot/assets/4165489/8cf16911-5bbf-4eb1-95a1-cdedc8556ccf)

**Comment by rubenslkirchner** [comment]
In my scenario the marker will be destroyed when x=0

**Comment by rubenslkirchner** [comment]
In my scenario the marker will be destroyed when x=0

**Comment by swharden** [comment]
> In my scenario the marker will be destroyed when x=0

Ah, that was the missing piece that confused me... I think this will be a good solution then! 

I plan on publishing a new package tonight so you can try it out 👍 

Thanks again for your feedback along the way!

**Comment by swharden** [comment]
I think this has been resolved by 72c0322 and 4378e83 but let me know if the behavior isn't what you expect and we can open this back up 👍

**Comment by rubenslkirchner** [comment]
Working perfectly!
